src/server/src/main/java/io/cassandrareaper/jmx/JmxConnectionFactory.java

```diff
@@ -86,31 +94,24 @@ public final JmxProxy connectAny(
       Collection<String> hosts,
       int connectionTimeout) throws ReaperException {
 
-    if (hosts == null || hosts.isEmpty()) {
-      throw new ReaperException("no hosts given for connectAny");
-    }
+    Preconditions.checkArgument(null != hosts && !hosts.isEmpty(), "no hosts provided to connectAny");
+
     List<String> hostList = new ArrayList<>(hosts);
     Collections.shuffle(hostList);
-    Iterator<String> hostIterator = hostList.iterator();
 
     for (int i = 0; i < 2; i++) {
-      while (hostIterator.hasNext()) {
-        String host = hostIterator.next();
-
+      for (String host : hostList) {
+        assert null != host; // @todo remove the null check in the following if condition
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)